src/main/java/org/folio/circulation/domain/CheckInProcessRecords.java

```diff
@@ -101,7 +107,18 @@ public CheckInProcessRecords withHighestPriorityFulfillableRequest(Request reque
       this.loan,
       this.requestQueue,
       this.checkInServicePoint,
-      request);
+      request, loggedInUserId);
+  }
+
+  public CheckInProcessRecords withLoggedInUserId(String userId) {
+    return new CheckInProcessRecords(
+        this.checkInRequest,
+        this.item,
+        this.loan,
+        this.requestQueue,
+        this.checkInServicePoint,
+        this.highestPriorityFulfillableRequest,
+        UUID.fromString(userId));
```

Comment:
What happens if userId is null?

Reply:
I don't think the checkin should be halted when there is no userId. Since the userId is taken from a system-wide header, it is assumed that this is safe.
A null check in another place in code.

Reply:
I don't think we should assume that the header is safe. What does the null check do? Does it fail the entire check in?

What are the requirements for when a deactivated / unknown user attempts a check in?

Reply:
No, it does not fail the check in, it simply does not put the userId field in the last checkin.

Reply:
answer: 
if user is unknown, the item record should just show a - in the Source field. However, it's still important to record the time and service point for that check in.

If the user is inactive but still able to complete a check in, then the system should still record them as the source. However, I'm not sure that can happen.

I believe, that is how it is implemented

Reply:
Is this describing how it is currently implemented, or what folks (users, SIG, PO etc) want the behaviour to be?

Reply:
Yes, there is an answer from PO above.